platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Func.kt

```diff
@@ -100,7 +100,13 @@ class FuncAdapter(
         req: InvokeFuncReq,
         responseHandler: (ExecInvokeSubmitted) -> T
     ): T {
+
         val func = funcQueryRepository.get(funcId)
+        val codeVersion = req.version?.let {
+            codeQueryRepository.get(func.code.id, it).version
+        } ?: func.code.version
+        //funcQueryRepository.get(funcId, codeVersion)?
```

Comment:
I don't understand this - what is this supposed to do?

Reply:
It should check if version is supplied in req, else proceed with latest version. Then verify the supplied version exists, there is a test for this.